packages/types/src/Metadata/v3/toV2.ts

```diff
@@ -2,9 +2,37 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import Option from '../../codec/Option';
+import Vector from '../../codec/Vector';
 import MetadataV2 from '../v2/Metadata';
 import MetadataV3 from '.';
+import { MetadataModule } from './Metadata';
+import { StorageFunctionMetadata } from './Storage';
 
-export default function toV2 (v2: MetadataV3): MetadataV2 {
-  return new MetadataV2(v2);
+function storageV2 (storage: Option<Vector<StorageFunctionMetadata>>): Option<Vector<StorageFunctionMetadata>> {
+  if (storage.isNone) {
+    return storage;
+  }
+
+  const storageV2 = storage.unwrap().filter(({ type }) => !type.isDoubleMap);
+
+  return new Option<Vector<StorageFunctionMetadata>>(
+    Vector.with(StorageFunctionMetadata),
+    new Vector<StorageFunctionMetadata>(StorageFunctionMetadata, storageV2)
+  );
+}
+
+function modulesV2 (v3: MetadataV3): Array<MetadataModule> {
+  return v3.modules.map((mod) =>
+    new MetadataModule({
```

Comment:
This is an additional issue I found and the one that I didn't want to debug completely, hence reverting - when constructing, you are losing the events & calls, i.e. https://github.com/polkadot-js/api/blob/master/packages/types/src/Metadata/v2/Metadata.ts#L22

Reply:
Thank you for the feedback! This is a tempory solution to prevent error when using v3 and double map at the same time. I can remove this if we won't use v3 and double map together.

Reply:
Otherwise, do we want to support double map in v3? The easiest way is converting v3 to v4.

Reply:
The issue is that we are not passing `events` though here, i.e. it should be something like -

```js
new MetadataModule({
      name: mod.name,
      prefix: mod.prefix,
      storage: storageV2(mod.storage),
      events: ... // missing currently
      calls: ... // missing currently
})
```

So basically when we convert, we don't have those populated

Reply:
Previously, there is an error if the metadata has a doublemap type and calling as v0. I change toV2 function to prevent that error. I see you have added [DoubleMapType to StorageFunctionType in v0](https://github.com/polkadot-js/api/blob/bd22da687e385983dca2b080b335204bcc4df60f/packages/types/src/Metadata/v0/Storage.ts#L82), which will prevent this error. So I think I can remove my changes about this toV2?